front/src/pages/AdminPages/Scheduling.tsx

```diff
@@ -1,7 +1,62 @@
+import WeeklyShiftCreatorPanel from "../../components/WeeklyShiftCreatorPanel.tsx";
+import {useState} from "react";
+import {Guid} from "guid-typescript";
+import {ShiftMetadata, ShiftTypes} from "../../components/ScheduleAndShiftsCreationComponents/Types.ts";
+import {ScheduleModel} from "@noadudai/scheduler-backend-client";
+import {useCreateNewShiftsSchedule} from "../../apis.ts";
+
 const Scheduling = () => {
+    const daysInWeek: number = 7;
+    const today: Date = new Date();
+    const nextSunday = daysInWeek - today.getDay();
+    const nextWeeksDayDates: Date[] = Array.from({length: daysInWeek}, (_, i) => new Date(today.getFullYear(), today.getMonth(), today.getDate() + nextSunday + i));
+
+    const initialState: ShiftMetadata[] = Array.from(Object.values(ShiftTypes), (type) => nextWeeksDayDates.map((date) => {
+        return ({id: Guid.create(), shiftType:type, startDateAndTime: date, endDateAndTime: undefined});
+    })).flat();
+
+    const [shiftsSchedule, setShiftsSchedule] = useState<ShiftMetadata[]>(initialState);
+    const [isCreatingShiftsOpen, setIsCreatingShiftsOpen] = useState(false);
+
+    const saveEditingShiftToSchedule = (shiftId: Guid, startDateAndTime: Date, endDateAndTime: Date) => {
+        setShiftsSchedule(prev =>
+            prev.map((shift) =>
+                shift.id === shiftId ? { ...shift, startDateAndTime: startDateAndTime, endDateAndTime: endDateAndTime } : shift));
+    };
+
+    const mutation = useCreateNewShiftsSchedule();
+
+    const submitShiftsSchedule = () => {
+        // Only the shifts that have a defined endDateAndTime, are shifts that the manager created for the schedule.
+        const shiftsForMutation = shiftsSchedule.filter((shift): shift is ShiftMetadata  & { endDateAndTime: Date }  => shift.endDateAndTime !== undefined);
```

Comment:
if you create `shiftsForMutation` outside of the submit. you can use it not being empty as for the saving button being active. double bubble